client-sdks/ai-sdk/src/transformers.ts

```diff
@@ -514,20 +528,50 @@ export function transformNetwork(
   payload: NetworkChunkType,
   bufferedNetworks: Map<
     string,
-    { name: string; steps: StepResult[]; usage: LanguageModelV2Usage | null; output: unknown | null }
+    {
+      name: string;
+      steps: (StepResult & {
+        id: string;
+        iteration: number;
+        task: null | Record<string, unknown>;
+        input: StepResult['input'];
+        [PRIMITIVE_CACHE_SYMBOL]: Map<string, any>;
+      })[];
+      usage: LanguageModelV2Usage | null;
+      output: unknown | null;
+    }
   >,
   isNested?: boolean,
 ): InferUIMessageChunk<UIMessage> | NetworkDataPart | DataChunkType | null {
   switch (payload.type) {
     case 'routing-agent-start': {
       if (!bufferedNetworks.has(payload.runId)) {
         bufferedNetworks.set(payload.runId, {
-          name: payload.payload.agentId,
+          name: payload.payload.networkId,
           steps: [],
           usage: null,
           output: null,
         });
       }
+
+      const current = bufferedNetworks.get(payload.runId)!;
```

Comment:
In the other codepaths you don't force it with `!` and return null if undefined. On purpose here?